phoenix-core/src/it/java/org/apache/phoenix/end2end/PhoenixRowTimestampFunctionIT.java

```diff
@@ -191,4 +192,158 @@ public void testRowTimestampColumn() throws Exception {
+                    + " CONSTRAINT PK PRIMARY KEY(PK1, PK2))" + this.tableDDLOptions;
+            conn.createStatement().execute(ddl);
+
+            String dml = "UPSERT INTO " + tableName + " (PK1, PK2, KV1, KV2) VALUES (?, ?, ?, ?)";
+
+            long rowTimestamp = EnvironmentEdgeManager.currentTimeMillis();
+            Date rowTimestampDate1 = new Date(rowTimestamp - 120000);
+            PreparedStatement stmt = conn.prepareStatement(dml);
+
+            int count = 5;
+            for (int id = 0; id < count; ++id) {
+                int idValue = id;
+                stmt.setInt(1, idValue);
+                stmt.setDate(2, rowTimestampDate1);
+                stmt.setString(3, "KV1_" + idValue);
+                stmt.setString(4, "KV2_" + idValue);
+                stmt.executeUpdate();
+            }
+            conn.commit();
+
+            // case: No rows should be selected
+            // Since we used a past date for column PK2
+            String dql1 = "SELECT PHOENIX_ROW_TIMESTAMP() FROM " + tableName +
+                    " WHERE PHOENIX_ROW_TIMESTAMP() = PK2";
+
+            ResultSet rs1 = conn.createStatement().executeQuery(dql1);
+            assertTrue(!rs1.next());
+            rs1.close();
+
+            // case: All rows selected should have the phoenix_row_timestamp() > date column
+            // Since we used a past date for column PK2
+            String dql2 = "SELECT PHOENIX_ROW_TIMESTAMP() FROM " + tableName +
+                    " WHERE PK2 < PHOENIX_ROW_TIMESTAMP() ";
+
+            ResultSet rs2 = conn.createStatement().executeQuery(dql2);
+            int actualCount2 = 0;
+            while(rs2.next()) {
+                assertTrue(rs2.getDate(1).after(rowTimestampDate1));
+                actualCount2++;
+            }
+            assertEquals(count, actualCount2);
+            rs2.close();
+
+            // case: All rows selected should have the phoenix_row_timestamp() < date column
+            // So using a future date.
+            Date rowTimestampDate2 = new Date(rowTimestamp + 120000);
+            for (int id = 0; id < count; ++id) {
+                int idValue = id + 100;
+                stmt.setInt(1, idValue);
+                stmt.setDate(2, rowTimestampDate2);
+                stmt.setString(3, "KV1_" + idValue);
+                stmt.setString(4, "KV2_" + idValue);
+                stmt.executeUpdate();
+            }
+            conn.commit();
+
+            String dql3 = "SELECT PHOENIX_ROW_TIMESTAMP() FROM " + tableName +
+                    " WHERE PHOENIX_ROW_TIMESTAMP() < PK2";
+
+            ResultSet rs3 = conn.createStatement().executeQuery(dql3);
+            int actualCount3 = 0;
+            while(rs3.next()) {
+                assertTrue(rs3.getDate(1).before(rowTimestampDate2));
+                actualCount3++;
+            }
+            assertEquals(count, actualCount3);
+            rs3.close();
+        }
+    }
+
 }
```

Comment:
@jpisaac  Since we rely on the empty column timestamp to be reflected as the `PHOENIX_ROW_TIMESTAMP()`, should we add tests to confirm this? Also perhaps add tests that confirm that this is modified whenever a row is updated

Reply:
The EmptyColumnIT tests do an exhaustive verification around that condition. Let me know if there is anything missing?